### ISSUE TITLE
Admin and public side "Search" buttons missing names

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2280]
Both the admin side "Search" button that allows searching by resource type as well as the public-facing full text site "Search" button appear to be missing accessible names and/or descriptions. A solution could be to mimic Classic, which correctly uses "Search Submit" as an accessible name for this kind of button.

**Comment by aviannamiller** [comment]
Generally speaking, the fix for the admin and public side search submit button looks great (tested with VoiceOver). However, the fixes do not appear to have applied to Foundation, Freedom, Cozy, or Lively, all of which are missing the updated `aria-label` for the search button. Foundation and Cozy do technically have accessible names ("Search", not the "Submit" `aria-label`), but Freedom and Lively are lacking a name entirely for the search button.

I am getting some redundancy in the readout from VoiceOver on the text input search box:
<img width="618" alt="Screenshot of search bar with VoiceOver reading out 'Search Query Search, menu pop up, edit text, Search form'" src="https://github.com/user-attachments/assets/387b77b7-b287-467b-b250-4df376986b0b" />
My best guess is that this is due to the presence of both `aria-label` and `aria-labelledby` on the element. When used together, `aria-labelledby` supersedes `aria-label`, so that may be what's happening here. Note that this specific issue does not impact Foundation, Freedom, Cozy, or Lively, though, as again it appears the fixes may not have reached those themes.

**Comment by kimisgold** [comment]
>  However, the fixes do not appear to have applied to Foundation, Freedom, Cozy, or Lively, all of which are missing the updated aria-label for the search button.

That's likely because those themes have some kind of custom version of the search form and aren't relying on the default view. I can file issues in those repositories to update their markup.

> I am getting some redundancy in the readout from VoiceOver on the text input search box:

Part of that redundancy is likely because of the placeholder, which is getting read out. If you replace it with anything else, Voiceover should read "Search Query [your query here]". I'm trying to think of a replacement to avoid that redundancy, maybe something like "Enter a search term" so that it's clearer what's being read out.

**Comment by kimisgold** [comment]
Okay, the core's search form has been updated with these revisions:

* The submit button now includes a font icon `<span>` with `role="presentation"` set to avoid screen readers accidentally reading the unicode content that sets the font icon.
* I've reverted the submit button label back to text content, but with the `sr-only` class that has it only read out by screen readers. If themes want a text label in addition to or to replace the font icon, it's easy to restyle to be visible.
* Placeholder now reads "Enter a search term" to reduce redundancy for screen readers.

All the themes should be updated with these changes.

**Comment by aviannamiller** [comment]
These changes look great across all themes (and also seem to resolve some search button issues I encountered with [The Daily](https://github.com/omeka-s-themes/thedaily/issues/54) and [Thanks, Roy](https://github.com/omeka-s-themes/thanksroy/issues/14)). The new placeholder text is thoughtfully done. There may always be varying levels of redundancy across screen readers and browsers (tested on Chrome, Firefox, and Safari), but this seems like a great solution. Eventually testing with NVDA and JAWS would be ideal.

**Comment by aviannamiller** [comment]
I am returning some new parsing errors that have now been picked up for Foundation post-search form revisions. There are three duplicate IDs:
- `search-form`
- `fulltext-search`
- `search-submit`

The first occurrences are around lines 64 to 66 on my end:
<img width="855" alt="Screenshot of page source with first occurrences of affected code, which is highlighted" src="https://github.com/user-attachments/assets/33af74fa-bfc6-485c-959a-f804c1cbd49b" />
Duplicates appear around lines 127 to 129:
<img width="868" alt="Screenshot of page source with repetitions of affected code, which is highlighted" src="https://github.com/user-attachments/assets/340f6898-c416-4a31-aa71-460916560757" />

**Comment by kimisgold** [comment]
Previously, Foundation was duplicating the search form with one instance in the desktop header bar and another in an offcanvas pane for mobile. I'm now using JS to move the search form between those two spaces when the mobile nav toggle is triggered.

I've also slightly adjusted the core search form pattern again to rely on an aria-label instead of button text. There is a bug where browsers will still read aria-hidden contents if they are children of a button. This results in screen readers reading out the unicode glyph we use for the font icon. Using an aria-label will prevent that glyph from being read out.

Pull latest changes from Omeka S's `develop` branch as well as Foundation's `master` branch.

**Comment by aviannamiller** [comment]
The updates to Foundation's search form look good, although parsing is now detecting `main-nav` duplicated as an ID in both the header bar and offcanvas menu.

The `aria-label` implementation behaves well with VoiceOver, though testing with other screen readers would be most advisable.

**Comment by aviannamiller** [comment]
Thanks, that fixes the ID issue.

**Comment by kimisgold** [comment]
Things were generally working in JAWS and NVDA, though I noticed that the advanced search options toggle in the admin search wasn't announcing an expanded and collapsed status. That should be fixed now.

I also removed used of `aria-labelledby` to chain labels. Both public and admin search forms should have `role="search"`, and that should be enough context for screen readers to associate with the individual element labels. Now the inputs should mainly read "Query", "Advanced options", and "Submit". When the user tabs out of the form and then navigates back, the screen reader should announce they've re-entered the search form before announcing the submit button information.

If you can reconfirm everything's working as it should in Voiceover, I think this should be ready to close.

**Comment by aviannamiller** [comment]
Thank you for the testing support. I am getting those updated inputs (and the expanded/collapsed state announcement for the toggle) with VO, so it seems safe to say the fixes are nicely carrying across as expected!
